sdks/python/apache_beam/io/textio_test.py

```diff
@@ -0,0 +1,299 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Tests for textio module."""
+
+import glob
+import gzip
+import logging
+import tempfile
+import unittest
+import zlib
+
+import apache_beam as beam
+import apache_beam.io.source_test_utils as source_test_utils
+
+# Importing following private classes for testing.
+from apache_beam.io.textio import _TextSink as TextSink
+from apache_beam.io.textio import _TextSource as TextSource
+
+from apache_beam.io.textio import ReadFromText
+from apache_beam.io.textio import WriteToText
+
+from apache_beam import coders
+from apache_beam.io.filebasedsource_test import EOL
+from apache_beam.io.filebasedsource_test import write_data
+from apache_beam.io.filebasedsource_test import write_pattern
+from apache_beam.io.fileio import CompressionTypes
+
+from apache_beam.transforms.util import assert_that
+from apache_beam.transforms.util import equal_to
+
+
+class TextSourceTest(unittest.TestCase):
```

Comment:
For these tests, do we want to increase the number of lines / size of the files used, as well as the size of the lines?  As is, it seems that there is only a single test for files exceeding the read buffer size.  This means that we are probably getting light test coverage of the buffer extending / truncation logic.